PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/AzureSettingPanel.java

```diff
@@ -142,6 +172,32 @@ protected void artifactActionPerformed(Artifact selectArtifact) {
         }
     }
 
+    protected void syncBeforeRunTasks(AzureArtifact azureArtifact,
+                                               final RunConfiguration configuration) {
+        if (!Comparing.equal(lastSelectedAzureArtifact, azureArtifact)) {
```

Comment:
Did we implements equals method in `AzureArtifact`?

Reply:
you can impl, but cannot be used here since it is hard to impl equals correctly without project